src/wp-includes/option.php

```diff
@@ -164,6 +164,11 @@ function get_option( $option, $default = false ) {
 		// Prevent non-existent options from triggering multiple queries.
 		$notoptions = wp_cache_get( 'notoptions', 'options' );
 
+		// Prevent non-existent `notoptions` key from triggering multiple key lookups.
+		if ( $notoptions === false ) {
+			wp_cache_set( 'notoptions', [], 'options' );
```

Comment:
Moving lines 200-202 to here, to set `$notoptions` to an array here would be of benefit.

```suggestion
		if ( ! is_array( $notoptions ) ) {
			$notoptions = array();
			wp_cache_set( 'notoptions', $notoptions, 'options' );
```